src/Concerns/HasLayouts.php

```diff
@@ -2,58 +2,166 @@
+use Plank\Contentable\Enums\LayoutMode;
+use Plank\Contentable\Enums\LayoutType;
+use Plank\Contentable\Exceptions\MissingLayoutException;
 
 /**
  * @mixin Model
  * @mixin Layoutable
  */
 trait HasLayouts
 {
+    public function layout(): Layout
+    {
+        if ($this->relatedLayout === null) {
+            throw MissingLayoutException::show(static::class, $this->getKey());
+        }
+
+        return $this->relatedLayout;
+    }
+
+    public static function indexLayout(): Layout
+    {
+        $layoutModel = static::layoutModel();
+
+        $layout = $layoutModel::query()
+            ->where($layoutModel::getLayoutKeyColumn(), static::indexLayoutKey())
+            ->first();
+
+        if ($layout === null) {
+            throw MissingLayoutException::index(static::class);
+        }
+
+        return $layout;
+    }
+
+    public function layouts(): Collection
+    {
+        $layoutModel = static::layoutModel();
+
+        return $layoutModel::query()
+            ->where($layoutModel::getLayoutableColumn(), static::layoutKey())
+            ->when($this->globalLayouts(), function (Builder $query) use ($layoutModel) {
+                $query->orWhere(function (Builder $query) use ($layoutModel) {
+                    $query->where($layoutModel::getTypeColumn(), LayoutType::Global)
+                        ->whereNotIn($layoutModel::getLayoutKeyColumn(), $this->excludedLayouts());
+                });
+            })
+            ->get()
+            ->sortBy($layoutModel::getNameColumn());
+    }
+
     /**
-     * {@inheritDoc}
+     * The default implementation ships as a parent relationship
+     *
+     * @return BelongsTo
      */
-    public function layout(): LayoutContract
+    public function relatedLayout()
     {
-        /** @var class-string<LayoutContract&Model> $model */
+        /** @var class-string<Layout&Model> $layoutModel */
         $layoutModel = config('contentable.layouts.model');
 
-        return $layoutModel::query()
-            ->where($layoutModel::getLayoutKeyName(), $this->layoutKey())
-            ->firstOrFail();
+        return $this->belongsTo(
+            $layoutModel,
+            $this->layoutForeignKey(),
+            $this->layoutOwnerKey(),
+        );
     }
 
     /**
-     * {@inheritDoc}
+     * The foreign key for the layout relationship
      */
-    public function layoutKey(): string
+    protected function layoutForeignKey(): string
     {
-        return str(class_basename($this))
-            ->plural()
-            ->lower()
-            ->append('.show');
+        return 'layout_id';
     }
 
     /**
-     * {@inheritDoc}
+     * The primary key on the layouts table
      */
-    public static function indexLayout(): LayoutContract
+    protected function layoutOwnerKey(): string
     {
-        /** @var class-string<LayoutContract&Model> $layoutModel */
-        $layoutModel = config('contentable.layouts.model');
+        return 'id';
```

Comment:
Does using `$this->getKeyName()` make sense here instead?
```suggestion
        return $this->getKeyName();
```

Reply:
In this case we would need to:

```
$layoutModel = static::layoutModel();

return (new $layoutModel)->getKey();
```

That felt worse to me at the time, but now that I look at it again, it does reduce the amount of things to consider for package consumers.

Will implement.